core/cpu.cpp

```diff
@@ -45,333 +51,950 @@ u8 CPU::exec(){
+            setHalfCarryFlag(true);
+            return 4;
+        }
+        case 0x3F: {
+            //CCF
+            setCarryFlag(!readCarryFlag());
+            setHalfCarryFlag(false);
+            setSubtractFlag(false);
+            return 4;
+        }
+        case 0xCD: {
+            //CALL a16
+            sp--;
+            sp--;
+            mmu.write(sp, pc);
+
+            
+            //PC NEEDS TO BE INCREMENTED TWICE ON 16 BIT READ
+            u16 nn_nn = mmu.read16Bit(pc++);
+            pc++;
+            pc = nn_nn;
+            break; 
+        }
+        case 0xFE: {
+            //CP d8
+            //CP is a subtraction from A that doesn't update A, only the flags it would have set/reset if it really was subtracted.
+            u8 n = mmu.read(pc++);
+            u8 a = getHighByte(af);
+            u8 compare = a - n;
+
+            setSubtractFlag(true);
+            setCarryFlag(compare < 0);
+            setZeroFlag(compare == 0);
+
+            //TODO: need to look up when to set half cary flag
             break;
-        
-        case 0x31:
-            //LD SP,d16
-        
-        case 0xE2:
+        }
+        case 0xE2: {
             //LD (C),A same as LD($FF00+C),A
             mmu.write(getLowByte(bc) + 0xFF00, getHighByte(af));
             // sleep?
             cycles = 8;
             break;
-        
-        case 0xEA:
+        }
+        case 0xEA: {
             //LD (a16),A
             break;
-        
-        case 0x77:
-            //LD (HL),A
-            mmu.write(hl, getHighByte(af));
-            // SOMETHING ABOUT A MEMORY SLEEP CYCLE???
-            cycles = 8;
-            break;
-        case 0x32:
-            //LD (HL-),A
-            mmu.write(hl--, getHighByte(af));
-            // SOMETHING ABOUT A MEMORY SLEEP CYCLE???
-            cycles = 8;
-            break;
-        case 0x22:
-            //LD (HL+),A
-            mmu.write(hl++, getHighByte(af));
-            // SOMETHING ABOUT A MEMORY SLEEP CYCLE???
-            cycles = 8;
-            break;
-
-        case 0xF0:
+        }
+        case 0xF0: {
             //LDH A,(a8) same as LD A,($FF00+a8)
             break;
-        case 0xE0:
+        }
+        case 0xE0: {
             //LDH (a8),A same as LD ($FF00+a8),A
 
             break;
-        
-        case 0xC5:
+        }
+        case 0xC5: {
             //PUSH BC
             break;
-        
-        case 0xC1:
+        }
+        case 0xC1: {
             //POP BC
             break;
-        
-        case 0xC9:
+        }
+        case 0xC9: {
             //RET
             break;
-        
-        case 0x17:
-            //RLA
-            break;
-        
-        case 0x90:
-            //SUB B
-            u8 b = getHighByte(bc);
+        } 
+        case 0x76: {
+            //HALT
+            std::cout << "Halt Instruction Reached" << std::endl;
+            std::exit(HALT_EXIT);
+            return 4;
+        }
```

Comment:
Oh, awesome. I didn't even reference the Pan docs for this one, and I totally misunderstood. I think I'll just put a `TODO` here because we still have to think through the interrupt stuff once the op codes are closer to being done.